src/runtime/containerd-shim-v2/stream_test.go

```diff
@@ -229,6 +227,9 @@ func TestIoCopy(t *testing.T) {
 			}
 		}
 
+		// need to make sure all go routines from ioCopy() have properly started
+		time.Sleep(time.Second)
```

Comment:
I don't think this is going to be reliable. Is there any way you could select (with a timeout) on a channel instead?

Reply:
Ok - I'm trying something different.
I'm adding some read/write checks before I try closing pipes. This adds synchronization between the threads: the reader will have to wait for the writer to be done, so doing this actually makes sure everything is initialized before we start closing stuff. And there is a timeout in checkFifoRead() that ensures we won't be waiting forever.

Hope this works better.